go/mysql/replication_status.go

```diff
@@ -25,17 +25,28 @@ import (
 
 // ReplicationStatus holds replication information from SHOW SLAVE STATUS.
 type ReplicationStatus struct {
+	// Position is the current position of the replica. For GTID replication implementations
+	// it is the executed GTID set. For file replication implementation, it is same as
+	// FilePosition
 	Position Position
 	// RelayLogPosition is the Position that the replica would be at if it
 	// were to finish executing everything that's currently in its relay log.
 	// However, some MySQL flavors don't expose this information,
 	// in which case RelayLogPosition.IsZero() will be true.
 	// If ReplicationLagUnknown is true then we should not rely on the seconds
 	// behind value and we can instead try to calculate the lag ourselves when
-	// appropriate.
-	RelayLogPosition      Position
-	FilePosition          Position
-	FileRelayLogPosition  Position
+	// appropriate. For MySQL GTID replication implementation it is the union of
+	// executed GTID set and retrieved GTID set. For file replication implementation,
+	// it is same as FileRelayLogPosition
+	RelayLogPosition Position
+	// FilePosition stores the position of the source tablets binary log
+	// upto which the SQL thread of the replica has run.
+	FilePosition Position
+	// FileRelayLogPosition stores the position of the source tablets binary log
+	// upto which the IO thread has read and added to the relay log
+	FileRelayLogPosition Position
+	// RelayLogFilePosition stores the position in the relay log file
+	RelayLogFilePosition  Position
```

Comment:
These two field names are confusing. It is going to be too easy to make a mistake while making changes in future.

Reply:
Yes, renaming the field `FileRelayLogPosition` to `RelayLogSourceBinLogEquivalentPosition`

Reply:
😆 
Guess that's the best we can do to avoid confusion.
I don't know whether binlog is one word - so Binlog vs BinLog
_goes to grep vitessio/vitess_
OK, we use `Binlog` almost everywhere except in a few test files. I know this is super-nitpicky, but could you please change it?

Reply:
Okay 👍